Remotion/Core/Tools/ToolRunner.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Configuration;
+using System.IO;
+using System.Reflection;
+using JetBrains.Annotations;
+using Remotion.Utilities;
+
+#if NET
+namespace Remotion.Tools
+{
+  /// <summary>
+  /// Base class for executing code that should run in a different base directory and or use a different config file.
+  /// </summary>
+  public abstract class ToolRunner
```

Comment:
Since the class uses settings that cannot be undone I think it is useful to include the `Tools` in the naming. This indicates a bit more that the class should only be used in tool projects.

Reply:
Since it's in the Tools assembly, it feels like a redudant argument to me. I don't mind a name that's specific / has a prefix, I'm just not sure "Tool" is the best prefix. Do you have something that expresses the intent more clearly? Tools doesn't really imply much in terms of isolation etc.

Reply:
Oh, and the `Base` suffix is needed either way.

Reply:
How about `AppContextRunnerBase`? or `ChangedAppContextRunnerBase`?

Reply:
CustomAppContextRunnerBase? (only downside is that "custom" is not specific)
IsolatedAppContextRunnerBase? (problably not because it's not isolated in the process)

"Changed" feels too much like data / communication.

Thoughts? (naming = hard)

Reply:
I prefer `Custom` over `Isolated` since it does not imply that the context is changed back.